pkg/meta/meta.go

```diff
@@ -31,9 +32,23 @@ import (
 	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
 )
 
-// AnnotationKeyExternalName is the key in the annotations map of a resource for
-// the name of the resource as it appears on provider's systems.
-const AnnotationKeyExternalName = "crossplane.io/external-name"
+const (
+	// AnnotationKeyExternalName is the annotation key used to indicate the
+	// 'external' name, or unique identifier, of a managed resource. The
+	// external name may differ from the managed resource name - i.e. the
+	// resource's metadata.name.
+	AnnotationKeyExternalName = "crossplane.io/external-name"
+
+	// AnnotationKeyExternalNamePending is the annotation key used to
+	// indicate that a resource's external name is pending assignment.
+	// Crossplane prefers to use deterministic external names where it can,
+	// but in some cases this is not possible - for example when an API
+	// generates and returns the name at create time. In such cases this
+	// annotation should be set before an external resource is created.
+	// Crossplane uses this annotation to ensure it creates at most one
+	// external resource.
+	AnnotationKeyExternalNamePending = "crossplane.io/external-name-pending"
```

Comment:
I am wondering if we could combine this PR with https://github.com/crossplane/crossplane-runtime/pull/280 by using an annotation like `crossplane.io/external-name-pending-since` given we also set timestamp as a value here.

Reply:
I can't see a clean way to combine the two annotations into one, if that's what you're thinking? In this case it's important that we write the annotation _immediately before_ calling create, in order to guarantee that we have the latest version of the managed resource before we create the external resource. In #280 we want to record the time that the external resource was successfully created, i.e. after the creation of the external resource.

Perhaps though we could combine them in some other way, i.e. rather than adding an `external-name-pending` annotation then removing it when the name was no longer pending, we could add an annotation that indicated a create was about to happen, then another one indicating it succeeded? 🤔